### PR TITLE
Adds utility to export countylookup results to CSV

### PR DESCRIPTION
**Pull request by tamaradement**
## PR Description

- Implemented the exportToCSV function to take the zipToCounty map and write the data to standard output in CSV format.
- Added a command-line flag to allow users to view the output for a single zip code.

For context: this pr completes the second step in an implementation that will eventually do the following:
- [X] Write a program that leverages the zip4 package to print all counties associated with a zip code. * Addressed in #2 
- [X] Output data to CSV format
- [ ] Implement a function to convert county codes to county names (optional). This could make data analysis easier.
- [ ] Use the output or a similar program to find differences between USPS and the [Smarty US ZIP Code API](https://www.smarty.com/docs/cloud/us-zipcode-api).

Tested Plan
 - [X] Ran the program with a valid tar file and observed output in the generated CSV.
 - [X] Checked error messages for scenarios with a missing tar file argument.
 - [X] Checked error messages for scenarios with an invalid tar file.
 
 Output in CSV:
 ```
 ZIP Code,County Numbers
91012,[037]
91341,[037]
03765,[009]
15074,[007]
19374,[029]
29605,[045]
52043,[043]
74149,[143]
78239,[029]
81138,[023]
19609,[011]
36123,[101]
36456,[013 039 035]
// ... more rows ...
```

## PR Checklist
* [no] **New automated tests have been written to the extent possible.**
* [yes, ran `go build ./cmd/countylookup` without incident ] **The code has been checked for structural/syntactic validity.**
	- AMI/application: a build was performed
	- terraform changes: "terraform plan" checked on every affected environment
* [n/a] **(If applicable) the code has been manually tested on our infrastructure.**
	- AMI/application: deployed an a test or dev environment
	- terraform changes: applied to test or dev environment
	- script: run against test or dev environment
* [n/a] **Likely failure points and new functionality have been identified and tested manually.**
	Examples:
	- Application manually run in a way that triggers any new branches
	- AMI logged into and changes verified from login shell
* [yes, see above] **Pull request description includes a description of all the manual steps performed to accomplish the above.**

To provide feedback on this template, visit https://docs.google.com/document/d/1YfTv7Amyop5G_8w1c2GJ_Mu-70L0KkZHhm9f9umDi3U/edit

### CODE REVIEW

**Review by jonahb — COMMENTED** [review]
The code looks great. A couple questions on the design:

I was thinking that the main scenario is a person wanting to see the counties for a ZIP code (e.g. Smarty missing a county for a ZIP). In this case, CSV isn't a bad format, but it might not be necessary. Are there other scenarios where CSV would be helpful? (This scenario might also call for a way to print counties for a single ZIP.)

Also, I'm wondering whether it'd be better to write the CSV to stdout. That would let the user view the output immediately, pipe it to `grep`, etc. without a second command. They could still save it to a file by redirecting (`countylookup >file.csv`). On the other hand, this wouldn't help them name the file as the current design does. What are your thoughts?

**Comment by tamaradement** [comment]
Thanks for the feedback! Considering the broader goal of finding disparities between USPS and Smarty data, writing to stdout makes a lot of sense to me. We will lose the automatic file naming we have, but the flexibility we gain I think will be worth it. We can also provide file-naming conventions through documentation if it seems necessary.

**Comment by jonahb** [comment]
Sounds good to me. Do you want to make that change in this PR?

**Comment by tamaradement** [comment]
Yes, I'll work on changes for this pr.

File: cmd/countylookup/main.go [hunk 1]
```diff
@@ -1,30 +1,70 @@
...
+	err = exportToCSV(zipToCounty)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to export to CSV: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func exportToCSV(zipToCounty map[string][]string) error {
+	writer := os.Stdout
+	csvWriter := csv.NewWriter(writer)
```
Comment (jonahb) [inline cmd/countylookup/main.go]: You could save a line of code without sacrificing much by "inlining" `writer` here: `csvWriter := csv.NewWriter(os.Stdout)`.

File: cmd/countylookup/main.go [hunk 2]
```diff
@@ -1,30 +1,70 @@
...
+	}
+
+	return nil
+}
+
+func printCountiesForZIP(zipToCounty map[string][]string, zip string) {
+	if counties, found := zipToCounty[zip]; found {
+		fmt.Printf("ZIP Code: %s, Counties: %v\n", zip, counties)
+	} else {
+		fmt.Printf("No counties found for ZIP Code: %s\n", zip)
```
Comment (jonahb) [inline cmd/countylookup/main.go]: This is a detail, but probably worth exiting nonzero if the ZIP isn't found, similar to `ls`. Better to exit at the top-level, so you'd have to return something or fold this logic into `main`.

**Review by jonahb — APPROVED** [review]
Looks good! I left a couple optional comments. (Sometimes we approve with comments if there's nothing too important to address.)

File: cmd/countylookup/main.go (hunk 2 above)
Comment (tamaradement) [reply cmd/countylookup/main.go]: In that case, I'm opting to collapse this logic into main.

**Review by jonahb — APPROVED** [review]
Looks good!

**Pull request merged**